Common/Data/Consolidators/RangeConsolidator.cs

```diff
@@ -0,0 +1,223 @@
+        /// </summary>
+        /// <param name="range">The constant value size of each bar, where the unit is the minimum price variation of the RangeBar's symbol</param>
+        /// <param name="selector">Extracts the value from a data instance to be formed into a <see cref="RangeBar"/>. The default
+        /// value is (x => x.Value) the <see cref="IBaseData.Value"/> property on <see cref="IBaseData"/></param>
+        /// <param name="volumeSelector">Extracts the volume from a data instance. The default value is null which does
+        /// not aggregate volume per bar.</param>
+        public RangeConsolidator(decimal range,
+            PyObject selector,
+            PyObject volumeSelector = null)
+            : base(selector, volumeSelector)
+        {
+            Range = range;
+            _firstTick = true;
+        }
+
+        /// <summary>
+        /// Update the current RangeBar being created with the given data.
+        /// </summary>
+        /// <param name="time">Time of the given data</param>
+        /// <param name="currentValue">Value of the given data</param>
+        /// <param name="volume">Volume of the given data</param>
+        protected override void UpdateBar(DateTime time, decimal currentValue, decimal volume)
+        {
+            if (currentValue > CurrentRangeBar.Low + 2 * RangeSize)
+            {
+                Rising(time, currentValue, volume);
+            }
+            else if (currentValue < CurrentRangeBar.High - (2 * RangeSize))
+            {
+                Falling(time, currentValue, volume);
+            }
+            else
+            {
+                CurrentRangeBar.Update(time, currentValue, volume);
+            }
+        }
+
+        /// <summary>
+        /// Checks if the current RangeBar being created has closed. If that is the case, it consolidates
+        /// the bar, resets the current RangeBar and stores the needed information from the last bar to
+        /// create the next bar
+        /// <see cref="UpdateBar(DateTime, decimal, decimal)"/>
+        /// </summary>
+        protected override void CheckIfBarIsClosed()
+        {
+            if (CurrentRangeBar != null && CurrentRangeBar.IsClosed)
+            {
+                OnDataConsolidated(CurrentRangeBar);
+                CurrentRangeBar = null;
+            }
+        }
+
+        /// <summary>
+        /// Creates a new bar with the given data
+        /// </summary>
+        /// <param name="data">The new data for the bar</param>
+        protected override void CreateNewBar(IBaseData data)
+        {
+            var currentValue = Selector(data);
+            var volume = VolumeSelector(data);
+            var open = currentValue;
+
+            if (_firstTick)
+            {
+                _minimumPriceVariation = SymbolPropertiesDatabase.FromDataFolder().GetSymbolProperties(data.Symbol.ID.Market, data.Symbol, data.Symbol.ID.SecurityType, "USD").MinimumPriceVariation;
```

Comment:
We could have the symbol properties DB in a static property or method. Look at `OptionStrategies.SymbolPropertiesDatabase`.